src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/create-subscription-modal/create-subscription-modal.component.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright Kamu Data, Inc. and contributors. All rights reserved.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the LICENSE file.
+ */
+
+import { ChangeDetectionStrategy, Component, inject, Input, OnInit } from "@angular/core";
+import { AbstractControl, FormBuilder, Validators } from "@angular/forms";
+import { NgbActiveModal } from "@ng-bootstrap/ng-bootstrap";
+import AppValues from "src/app/common/values/app.values";
+import {
+    DatasetBasicsFragment,
+    WebhookSubscriptionInput,
+    WebhookSubscriptionStatus,
+} from "src/app/api/kamu.graphql.interface";
+import { WebhooksService } from "src/app/services/webhooks.service";
+import { BaseComponent } from "src/app/common/components/base.component";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import {
+    CreateWebhookSubscriptionSucces,
+    SubscribedEventType,
+    WebhookSubscriptionModalAction,
+} from "./create-subscription-modal.model";
+import { MaybeUndefined } from "src/app/interface/app.types";
+
+@Component({
+    selector: "app-create-subscription-modal",
+    templateUrl: "./create-subscription-modal.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CreateSubscriptionModalComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
+    @Input() public subscriptionData: CreateWebhookSubscriptionSucces;
+
+    public DROPDOWN_LIST: SubscribedEventType[] = [];
+    public secret: MaybeUndefined<string>;
+    public readonly WebhookSubscriptionStatus: typeof WebhookSubscriptionStatus = WebhookSubscriptionStatus;
+    public readonly WebhookSubscriptionModalAction: typeof WebhookSubscriptionModalAction =
+        WebhookSubscriptionModalAction;
+
+    public activeModal = inject(NgbActiveModal);
+    private fb = inject(FormBuilder);
+    private webhooksService = inject(WebhooksService);
+
+    public ngOnInit(): void {
+        this.webhooksService
+            .eventTypes()
```

Comment:
This might be cached. Never changes at runtime. Only after server restart maybe in future.